File: src/main/java/com/dev/farmmanager/service/auth/AuthService.java
package com.dev.farmmanager.service.auth;

import com.dev.farmmanager.domain.payload.auth.CreateUserPayload;
import com.dev.farmmanager.domain.payload.auth.EmailLoginPayload;
import lombok.NonNull;

public interface AuthService {

    String register(@NonNull final CreateUserPayload payload);

    String login(@NonNull final EmailLoginPayload payload);

}

File: frontend/src/services/auth.service.ts
import http from '@/api/http'
import type { LoginPayload, CreateUserPayload, AuthUser } from '@/types/auth'

export const authService = {
  login: (payload: LoginPayload) =>
    http.post('/auth/login', payload),

  register: (payload: CreateUserPayload) =>
    http.post('/auth/register', payload),

  logout: () =>
    http.post('/auth/logout'),

  me: () =>
    http.get<AuthUser>('/farm/user/me').then((r) => r.data),
}
